lib/rules/no-lonely-if.js

```diff
@@ -52,9 +51,26 @@ module.exports = {
                                 return null;
                             }
 
+                            if (
+                                node.consequent.type !== "BlockStatement" && lastIfToken.value !== ";" && tokenAfterElseBlock &&
+                                (
+                                    node.consequent.loc.end.line === tokenAfterElseBlock.loc.start.line ||
+                                    /^[(\[+-`]/.test(tokenAfterElseBlock.value) ||
+                                    lastIfToken.value === "++" ||
+                                    lastIfToken.value === "--"
+                                )
+                            ) {
```

Comment:
A non-related thing, we have several rules which convert a block to a non-block, so maybe it can extract this check (or something like) to `ast-utils.js`.